Check DBus loading status before attempting to detect its version

deralmas opened the pull request:
Fixes two related segfaults caused by running the DBus version check unconditionally and potentially calling null function pointers.

This was clearly an oversight as all the other wrappers have proper loading status checks.

## Extra notes

Note that I've tested this on my Wayland branch rebased onto master (with also some tasty PR "ports", pushing soon BTW), if for whatever reason the thing turns out to matter.

~~I'd also have implemented this with some early returns (which would've also made the diff smaller) but the code style around the part I touched seemed to indicate a preference for big if blocks. Just letting you know.~~ **Update**: Switched to early returns.

akien-mga commented:
> I'd also have implemented this with some early returns (which would've also made the diff smaller) but the code style around the part I touched seemed to indicate a preference for big if blocks.

I'd suggest going with early returns anyway, it's the preferred style when possible.

deralmas committed:
Check DBus loading status before attempting to detect its version

Fixes two related segfaults caused by running the DBus version check
unconditionally and potentially calling null function pointers.

This was clearly an oversight as all the other wrappers have proper
loading status checks.

akien-mga commented:
Cherry-pick not relevant as this code is new in 4.1.
